controllers/sr_controller/sr/robot/ruggeduino.py

```diff
@@ -17,36 +17,47 @@ def init_ruggeduino_array(webot):
         "left finger sensor",
         "right finger sensor",
     ]
+    led_names = [
+        "led 1",
+        "led 2",
+        "led 3",
+        "led 4",
+        "led 5",
+        "led 6",
+    ]
+
+    analogue_input_array = [DistanceSensor(webot, name) for name in dist_sensor_names]
 
-    analogue_array = [DistanceSensor(webot, name) for name in dist_sensor_names]
+    digital_input_array = [Microswitch(webot, name) for name in switch_names]
 
-    digital_array = [Microswitch(webot, name) for name in switch_names]
+    digital_output_array = [Led(webot, name) for name in led_names]
 
-    return [Ruggeduino(webot, analogue_array, digital_array)]
+    return [Ruggeduino(webot, analogue_input_array, digital_input_array, digital_output_array)]
 
 
 class Ruggeduino(object):
 
-    DIGITAL_PIN_OFFSET = 2  # Exclude pins 0 and 1 as they are used for USB serial comms
+    DIGITAL_READ_PIN_START = 2
 
-    def __init__(self, webot, analogue_array, digital_array):
+    def __init__(self, webot, analogue_input_array, digital_input_array, digital_output_array):
         self.webot = webot
-        self.analogue_array = analogue_array
-        self.digital_array = digital_array
+        self.analogue_input_array = analogue_input_array
+        self.digital_input_array = digital_input_array
+        self.digital_output_array = digital_output_array
 
     def digital_read(self, pin):
         "Read an digital input"
-        if pin < 2 or pin > 13:
-            raise ValueError("Only pins 2 - 13 are available on the Ruggeduino")
-        return self.digital_array[pin - Ruggeduino.DIGITAL_PIN_OFFSET].read_value()
+        return self.digital_input_array[pin - Ruggeduino.DIGITAL_READ_PIN_START].read_value()
```

Comment:
This interface is a little weird. On a real board the digital pins are the same list, and you can't have a sensor and LED in the same pin.

It'd be nice to offset these so the LED pins start after the switches.

Alternatively, creating some generic interface for a digital device, and having them implementing read / write. But that is a little more work than a basic offset and check

Reply:
> It'd be nice to offset these so the LED pins start after the switches.

This *is* how it currently works. Pins 2-6 are switches and pins 7-12 are LEDs